ARE/components/actuator.tooltip/src/main/java/eu/asterics/component/actuator/tooltip/GUI.java

```diff
@@ -67,139 +67,161 @@
     public GUI(final TooltipInstance owner) {
         super("TooltipPanel");
 
+        this.owner = owner;
         GraphicsDevice gd = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice();
-        int width = gd.getDisplayMode().getWidth();
-        int height = gd.getDisplayMode().getHeight();
-        Dimension dim = new Dimension(width, height);
-
-        screenWidth = (int) dim.getWidth();
-        screenHeight = (int) dim.getHeight();
+        screenWidth = gd.getDisplayMode().getWidth();
+        screenHeight = gd.getDisplayMode().getHeight();
 
         setUndecorated(true);
-        setAlwaysOnTop(true);
         setBackground(new Color(0, 0, 0, 0)); // transparent !
-        setSize(dim);
-        width = dim.width;
-        height = dim.height;
         setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
         setOpacity(0.5f);
-        setVisible(true);
-        setLocation(0, 0);
-        repaintInternal();
     }
 
+    /**
+     * Return currently active Tooltip filename.
+     * 
+     * @return
+     */
     String getTooltipFilename() {
         return actImageFileName;
     }
 
+    /**
+     * Tell if Tooltip is currently active (visible)
+     * 
+     * @return
+     */
     boolean tooltipsActive() {
         return tooltipActive;
     }
 
-    void activateTooltips(String tooltipFolder, int startIndex) {
-        this.tooltipFolder = tooltipFolder + "/";
-        this.tooltipStartIndex = startIndex;
-        actTooltip = tooltipStartIndex;
-        this.tooltipActive = true;
+    /**
+     * Activates the current Tooltip image. Sends an etpTooltipActivated afterwards.
+     */
+    void activateTooltips() {
+        if (actTooltip < 0) {
+            actTooltip = owner.propTooltipStartIndex;
+        }
         loadImage(actTooltip);
+        owner.etpTooltipActivated.raiseEvent();
+        this.tooltipActive = true;
     }
 
+    /**
+     * Deactivates the current Tooltip image. Sends an etpTooltipDeactivated afterwards.
+     */
     void deactivateTooltips() {
-        actTooltip = tooltipStartIndex;
-        this.tooltipActive = false;
         resetImage();
-        repaintInternal();
-    }
-
-    void navigateTooltips(float dx) {
-        if (System.currentTimeMillis() - tooltipTime > 100) {
-            if (dx > 0)
-                actTooltip++;
-            else {
-                actTooltip--;
-            }
-            loadImage(actTooltip);
-        }
+        owner.etpTooltipDeactivated.raiseEvent();
+        showTooltip(false);
+        this.tooltipActive = false;
     }
 
+    /**
+     * Selects the next index of Tooltip images. If the @see {@link #TOOLTIP_INDEX_MAX} is reached, starts with {@link #TOOLTIP_INDEX_MIN}.
+     */
     void navigateNextTooltip() {
-        navigateTooltips(1);
+        actTooltip = (actTooltip + 1 > TOOLTIP_INDEX_MAX) ? TOOLTIP_INDEX_MIN : actTooltip + 1;
```

Comment:
the intended and previously implemented functionality was, that a navigation out of the bounds of existing .png files does deactivate the tooltips which isn't the case anymore now. In this implementation the user cannot leave the tooltip mode without providing an extra command/event to deactivate tooltips, but users should be able to deactivate tooltips by the same actions that they are doing for navigating through the tooltips.